zproject/local_settings_template.py

```diff
@@ -45,27 +45,30 @@
 # SSO_APPEND_DOMAIN = "example.com")
 SSO_APPEND_DOMAIN = None
 
-# Configure the outgoing SMTP server below. The default configuration
-# is prepopulated for GMail servers.  Change EMAIL_HOST for other
-# hosts, or leave it unset or empty to skip sending email.  Note if
-# you are using a GMail account to send outgoing email, you will
-# likely need to configure that account as "less secure" here:
-# https://support.google.com/accounts/answer/6010255.
+# Configure the outgoing SMTP server below. You will need to complete
+# the following steps:
```

Comment:
It's required for password authentication in addition to things like missed message emails.  

So if you're using Google or LDAP auth and thus aren't using Zulip passwords and you don't need missed message notifications you could do without it.